src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/Handlers/AbstractEvaluationCommandLineHandler.cs

```diff
@@ -165,19 +165,27 @@ private void ApplyChangesToContext(IComparable version, IProjectChangeDiff diffe
         private void RemoveFromContextIfPresent(string includePath)
         {
             string fullPath = _project.MakeRooted(includePath);
-            if (_paths.Remove(fullPath))
+
+            // Remove from the context first so if Roslyn throws due to a bug 
+            // or other reason, that our state of the world remains consistent
+            if (_paths.Contains(fullPath))
             {
                 RemoveFromContext(fullPath);
+                Assumes.True(_paths.Remove(fullPath));
```

Comment:
It doesn't have conditional attribute but it has debuggerstepthrough and apparently it has an attribute that this code won't be affected by servicing releases and hence can be ngened etc. Might be better to just have the return value in a variable and call Assumes.True on that.